.github/workflows/lib_compat_test.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+
+# Verifies a binary uses only dynamic symbols from whitelisted library versions.
+# Prints the disallowed symbols and their versions on failure.
+# Usage: lib_compat_test.py bin/clangd --lib=GLIBC_2.18
+
+import argparse
+import subprocess
+import sys
+
+parser = argparse.ArgumentParser()
+parser.add_argument('binary')
+parser.add_argument('--lib', action='append', default=[],
+                    help="Whitelist a library, e.g. GLIBC_2.18 or GLIBC")
+parser.add_argument('--sym', action='append', default=[],
+                    help="Whitelist a symbol, e.g. crc32")
+args = parser.parse_args()
+
+# Parses GLIBC_2.3 into ("GLIBC", [2,3])
+# Parses GLIBC into ("GLIBC", None)
+def parse_version(version):
+  parts = version.split('_', 1)
+  if len(parts) == 1:
+    return (version, None)
+  return (parts[0], [int(p) for p in parts[1].split('.')])
```

Comment:
this will throw if `int(p)` fails. you seem to be relying on this returning None for non-numeric versions in accept_lib.

Reply:
there aren't any non-numeric versions, we want it to fail and I didn't think it was worth polishing the error message. But I'll improve it a bit.

Reply:
actually you're right, returning `(version, None)` is probably better than failing outright here.